backend/pkg/logger/data/logger.go

```diff
@@ -0,0 +1,115 @@
+package data
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/logger"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/packet/data"
+)
+
+type ValueName string
+
+const (
+	name abstraction.LoggerName = "data"
+)
+
+type Logger struct {
+	running    bool
+	lock       sync.Mutex
+	valueFiles map[data.ValueName]*os.File
+}
+
+type DataRecord struct {
+	packet *data.Packet
+}
+
+func (data *DataRecord) Name() abstraction.LoggerName {
+	return name
+}
+
+func (sublogger *Logger) Start() error {
+	sublogger.lock.Lock()
+	defer sublogger.lock.Unlock()
+
+	if sublogger.running {
+		fmt.Printf("Logger already running")
+		return nil
+	}
+	sublogger.newFiles()
+
+	sublogger.running = true
+	fmt.Printf("Logger started")
+	return nil
+}
+
+type num interface {
+	Value() float64
+}
+
+func (sublogger *Logger) PushRecord(record abstraction.LoggerRecord) {
+	valueMap := record.(*DataRecord).packet.GetValues()
+
+	for key, value := range valueMap {
+		dataPointer := record.(*DataRecord).packet
+
+		var val string
+
+		switch v := value.(type) {
+		case num:
+			val = strconv.FormatFloat(v.Value(), 'f', -1, 64)
+
+		case data.BooleanValue:
+			val = strconv.FormatBool(v.Value())
+
+		case data.EnumValue:
+			val = string(v.Variant())
+		}
+		sublogger.valueFiles[key].WriteString(dataPointer.Timestamp().Format(time.RFC3339) + "," + val)
+	}
+
+}
+
+func PullRecord(request abstraction.LoggerRequest) (abstraction.LoggerRecord, error) {
+	panic("TODO!")
+}
+
+func Stop(sublog *Logger) {
+	sublog.lock.Lock()
+	defer sublog.lock.Unlock()
+
+	if !sublog.running {
+		fmt.Printf("Logger already stopped")
+		return
+	}
+
+	sublog.running = false
+	fmt.Printf("Logger stopped")
+}
+
+func (sublogger *Logger) newFiles() error {
+	setTime := time.Now().Format(time.RFC3339) + ".csv"
+
+	numericFile, err := os.Create("/numeric/numeric_" + setTime)
+	if err != nil {
+		return logger.NewErrCreatingFile("numeric", time.Now())
+	}
+	sublogger.valueFiles["numeric"] = numericFile
+
+	booleanFile, err := os.Create("/boolean/boolean_" + setTime)
+	if err != nil {
+		return logger.NewErrCreatingFile("boolean", time.Now())
+	}
+	sublogger.valueFiles["boolean"] = booleanFile
+
+	enumFile, err := os.Create("/enum/enum_" + setTime)
+	if err != nil {
+		return logger.NewErrCreatingFile("enum", time.Now())
+	}
+	sublogger.valueFiles["enum"] = enumFile
+	return nil
```

Comment:
you are mixing up value type and value name. `numeric`, `boolean`, and `enum` are not the names of the values, the names are defined in the [ADE](https://docs.google.com/spreadsheets/d/1pCIzf-31UmcRl9FgIZc0U9V3A5ktl-yKpfL-282Rpuc/edit#gid=2061870537) for each measurement and packet.

At first the logger does not know the value names and it should create files as it finds them, when the `PushRecord` method gets a value with a name not present in the file map, call a method to create a file for it. With this you have the problem of getting values on different timestamps, for that when you call start store the start timestamp on the `data.Logger` struct and then use it to get the file timestamp